Install script review

gbenhaim opened the pull request:
Hi all,

Please review the install script.

Signed-off-by: gbenhaim galbh2@gmail.com

gbenhaim committed:
Install script review

Signed-off-by: gbenhaim <galbh2@gmail.com>

eedri commented:
can you change the name of the function "run_system_tests" to "run_suite" or something else,
since we want it to run any supported suite, not necessarily system tests..

mykaul commented on install_scripts/install_lago.sh:
```diff
@@ -1,5 +1,6 @@
 #!/bin/bash -xe
 
+#Let's reviews the install script
```
reviews -> review

david-caro commented on install_scripts/install_lago.sh:
```diff
@@ -1,5 +1,6 @@
+#Let's reviews the install script
 
 REPONAME="ovirt-system-tests"
 REPOURL="https://gerrit.ovirt.org"
```
I'll add the comments here, as github does not allow commenting on the 'hidden' code :)
- the usage of 'function' for bash function is not recommended, as it's not as compatible as the other syntax (I think that old bash versions and other shells, not sure if it's an issue though)
- Use `grep -q` instead of just `grep`, as that will not show any output.
- +1 for the usage of local :wine_glass: 
- To configure the repos, you wight want to check the repomanage utilities, though it might not be supported on old el
- Missing a couple of quoted variable (specially $HOME and other non-locally generated ones)
- Hmmm, adding a user to a group will not be enough without relogging in again, you might want to run sudo or similar after (newgrp might be useful there too...)
- Also you will have to rmmod + insmod the intel_kvm after enabling nested, or reboot (rebooting will force the user to update the groups permissions too), maybe if it has to configure it, just showing a message with 'reboot needed please' will be enough.
- If the user is not root, you might need to sudo to chnge groups and enable/install stuff.

Great initiative!

gbenhaim replied:
Because the script must run under root, I don't see how 'newgrp' can help here.
Maybe I'll run "su - $username" at the end of the script ? (I'm getting the username as a positional parameter)

gbenhaim committed:
Refactored the install script

According to the review, I made some changes
to the install script:

1. Add support fo enabeling virtualization on amd
   systems.

2. Use of 'grep -q' instead of redirecting to /dev/null

3. Quoted variables.

4. System tests will be run under the user account, and
   not under root account.

5. Removed 'function' keyword from the functions.

6. Changed the name of 'run_system_tests' function to 'run_suite'.

Signed-off-by: gbenhaim <galbh2@gmail.com>

gbenhaim committed:
Change permission assignment to root's home dir

If the user who runs lago is root, execution permission
for a group should be assigned to /root.

Signed-off-by: gbenhaim <galbh2@gmail.com>

lenoch commented on install_scripts/install_lago.sh:
```diff
@@ -47,7 +57,7 @@ enabled=1
 
-    if ! grep "Fedora" /etc/redhat-release; then
+    if ! grep -q "Fedora" /etc/redhat-release; then
         cat > /etc/yum.repos.d/epel.repo <<EOF
```
yum -y install epel-release worked for me on a clean install of CentOS 7.3.
